### PR TITLE
Add support to have sample driver profile ebpf programs

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add support to have sample driver profile ebpf programs

Resolves: #462 

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: tests/sample/ext/drv/sample_ext.h [hunk 1]
```diff
@@ -54,4 +55,8 @@ sample_ebpf_extension_hook_provider_unregister();
  * @retval EBPF_OPERATION_NOT_SUPPORTED Operation not supported.
  */
 ebpf_result_t
-sample_ebpf_extension_invoke_program(_In_ const sample_program_context_t* context, _Out_ uint32_t* result);
\ No newline at end of file
+sample_ebpf_extension_invoke_program(_In_ const sample_program_context_t* context, _Out_ uint32_t* result);
+
+ebpf_result_t
```
Comment (dthaler) [inline tests/sample/ext/drv/sample_ext.h:60]: add doxygen comment

File: tests/sample/ext/drv/sample_ext.h [hunk 2]
```diff
@@ -54,4 +55,8 @@ sample_ebpf_extension_hook_provider_unregister();
...
+sample_ebpf_extension_profile_program(
+    sample_ebpf_ext_profile_request_t* request, size_t request_length, sample_ebpf_ext_profile_reply_t* reply);
```
Comment (dthaler) [inline tests/sample/ext/drv/sample_ext.h:62]: add sal

**Review by dthaler — APPROVED** [review]

**Comment by shankarseal** [comment]
If the plan is for performance test to load a sample program of sample_ext type and invoke it, please add such a test case.

**Review by shankarseal — COMMENTED** [review]
If the plan is for performance test to load a sample program of sample_ext type and invoke it, please add that support as part of this PR.

**Comment by Alan-Jowett** [comment]
> If the plan is for performance test to load a sample program of sample_ext type and invoke it, please add that support as part of this PR.

I was planning on adding that as a separate PR to keep the size of PR's manageable.

**Comment by Alan-Jowett** [comment]
Filed #594 to track updating performance tests to use this.

**Review by shankarseal — APPROVED** [review]

**Pull request merged**